File: src/main/java/com/example/wizytydomowe/Doctor/Doctor.java
package com.example.wizytydomowe.Doctor;

import lombok.*;

import javax.persistence.*;
import javax.validation.constraints.Digits;
import javax.validation.constraints.Email;

@Entity
@Data
@Table(name = "doctor")
public class Doctor {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;
    private String name;
    private String surname;
    @Digits(integer = 11, fraction = 0)
    private String phoneNumber;
    private String specialization;
    @Email
    private String email;
}

File: client/src/app/doctor.ts
export class Doctor {
  id: number = 1;
  name: string = '';
  surname: string = '';
  phoneNumber: string= '';
  specialization: string= '';
  email: string= '';

  constructor(id: number, name: string, surname: string, phoneNumber: string, specialization: string, email: string) {
    this.id = id;
    this.name = name;
    this.surname = surname;
    this.phoneNumber = phoneNumber;
    this.specialization = specialization;
    this.email = email;
  }

}

